src/student_registry_project/implementations/student_registry.rs

```diff
@@ -37,4 +50,46 @@ impl StudentRegistry {
     pub fn get_student_by_id(&self, id: u32) -> Option<&Student> {
         self.total_students.get(id as usize)
     }
+
+    // pub fn get_courses(&self) -> &Vec<Course> {
+        
+    // }
+
+    // Method to enroll a student in a course
+    pub fn enroll_student_in_course(&mut self, student_id: u32, course_id: u32) {
+        // Find the student and course
+        // let student_index = self.total_students.iter().position(|s| s.id == student_id).unwrap();
+        // let course_index = self.course_registry.iter().position(|c| c.id == course_id).unwrap();
+        let student_index: usize = (student_id - 1).try_into().unwrap();
+        let course_index: usize = (course_id - 1).try_into().unwrap();
+
+        // Check if the student is already enrolled in the course
+        if self.total_students[student_index].enrolled_course_ids.contains(&course_id) {
+            println!("Student is already enrolled in the course.");
+            return;
+        }
+
+        // Check if the course exists
+        if course_index >= self.course_registry.len() {
+            println!("Course not found.");
+            return;
+        }
+
+        // Check if the student exists
+        if student_index >= self.total_students.len() {
+            println!("Student not found.");
+            return;
+        }
+
+        // Check if the course has available capacity
+        if self.course_registry[course_index].enrolled_student_ids.len() < self.course_registry[course_index].capacity as usize {
+            // Add the student to the course's enrolled students
+            self.course_registry[course_index].enrolled_student_ids.push(student_id);
+
+            // Add the course to the student's enrolled courses
+            self.total_students[student_index].enrolled_course_ids.push(course_id);
+        } else {
+            println!("Course is full. Cannot enroll student.");
+        }
```

Comment:
Consider handling edge cases where the Vec indexes can shift, to avoid referencing to invalid or incorrect data.